### PR TITLE
Remove STL dependency

### PR DESCRIPTION
**Pull request by Gargaj**
For projects that have their own allocator; this uses imgui's memory allocator (which may be overridden).

Also, clean up inconsistent whitespaces. :)

### CODE REVIEW

File: src/app/imgui_remote.h [hunk 1]
```diff
@@ -15,485 +15,551 @@
...
+					if ( sscanf( (char *) data, "ImKeyPress=%d", &key ) == 1 )
+						ImGui::GetIO().AddInputCharacter( ImWchar( key ) );
 				}
-				else if (strstr((char *)data, "ImClipboard="))
+				else if ( strstr( (char *) data, "ImClipboard=" ) )
 				{
-					char *clipboard = &((char *)data)[strlen("ImClipboard=")];
-					ImGui::GetIO().SetClipboardTextFn(nullptr, clipboard);
+					char *clipboard = &( (char *) data )[ strlen( "ImClipboard=" ) ];
+					ImGui::GetIO().SetClipboardTextFn( clipboard );
```
Comment (Gargaj) [inline src/app/imgui_remote.h]: This seems to have changed in later versions.

**Comment by ocornut** [comment]
Hello,
I am not the maintainer of this project but I would say the PR is extremely noisy (unrelated changes of formatting, spaces etc.). I would suggest to compare your core with master ignoring white spaces and generating a new cleaner PR to support this merge.

**Comment by Gargaj** [comment]
> I am not the maintainer of this project but I would say the PR is extremely noisy (unrelated changes of formatting, spaces etc.). I would suggest to compare your core with master ignoring white spaces and generating a new cleaner PR to support this merge.

The reason it's noisy is because I unified the whitespaces; if you look at the original file, the indentation style is all over the place.

**Comment by ocornut** [comment]
I would suggest doing in two separate commits. Anyway it's not of me to say!

**Comment by JordiRos** [comment]
I would have preferred two separate commits indeed, it's easier to review and learn what has changed and could probably merge without testing if all seems ok, I will test and merge when I get some time this week.
Fixing indentation is nice thanks. Adding whitespaces in functions is subjective and I never do that, but I don't mind if that goes in.

**Comment by Gargaj** [comment]
> Adding whitespaces in functions is subjective and I never do that, but I don't mind if that goes in.

I can revert that, was probably just my VS autoformat settings being overzealous.

File: src/app/imgui_remote.h (hunk 1 above)
Comment (JordiRos) [reply src/app/imgui_remote.h]: Sorry for the delay, looking good! Thanks for your contribution

**Pull request merged**